charms/knative-operator/src/charm.py

```diff
@@ -50,12 +50,16 @@ def __init__(self, *args):
         self._observability_resource_handler = None
 
         # Instantiate MetricsEndpointProvider for Prometheus scraping
+        targets = ["*:9090"]
         if self._otel_exporter_ip:
-            self._scraping = MetricsEndpointProvider(
-                self,
-                relation_name="metrics-endpoint",
-                jobs=[{"static_configs": [{"targets": [f"{self._otel_exporter_ip}:8889"]}]}],
-            )
+            targets.append(f"{self._otel_exporter_ip}:8889")
+
+        self._scraping = MetricsEndpointProvider(
```

Comment:
Should we keep this consistent among charms? I just noticed that previously we were using[prometheus_provider](https://github.com/canonical/istio-operators/pull/477/files#diff-d4a88abbd5dcb0911fc960c33c9f0440bff32b2ddc60d56d266788300b709e02R38).